src/main/java/org/reaktivity/rym/internal/RymInstall.java

```diff
@@ -0,0 +1,226 @@
+        {
+            if (!dependenciesEl.isJsonObject())
+            {
+                // TODO Handle file not found
+                throw new JsonSyntaxException(String.format("%s is not a JSON object", PROPERTY_DEPENDENCIES));
+            }
+            JsonObject dependenciesObj = (JsonObject)dependenciesEl;
+            dependenciesObj.keySet().forEach(key ->
+            {
+                String artifact = key.indexOf(':') > 0 ? key : String.format("%s:%s", DEFAULT_GROUP_NAME, key);
+                JsonElement versionEl = dependenciesObj.get(key);
+                // TODO Process version here. e.g. Is it a range?
+                String version = versionEl.isJsonPrimitive() ? versionEl.getAsString() : versionEl.toString();
+                dependencies.put(artifact, version);
+            });
+        }
+    }
+
+    private void writeDepsLockFile() throws IOException
+    {
+        JsonObject depsLockObj = new JsonObject();
+
+        JsonObject repositoriesObj = new JsonObject();
+        depsLockObj.add(PROPERTY_REPOSITORIES, repositoriesObj);
+        repositories.forEach((n, u) -> repositoriesObj.add(n, new JsonPrimitive(u)));
+
+        if (dependencies.size() > 0)
+        {
+            JsonObject dependenciesObj = new JsonObject();
+            depsLockObj.add(PROPERTY_DEPENDENCIES, dependenciesObj);
+            dependencies.forEach((dep, ver) ->
+            {
+                dependenciesObj.add(dep, new JsonPrimitive(ver));
+            });
+        }
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        try (BufferedWriter br = new BufferedWriter(new FileWriter(DEPENDENCY_LOCK_FILENAME)))
+        {
+            br.write(gson.toJson(depsLockObj));
+        }
+
+    }
+
+    private boolean resolveDependencies(
+        ResolveOptions options) throws ParseException, IOException
+    {
+        ChainResolver chain = new ChainResolver();
+        chain.setName("default");
+
+        repositories.forEach((n, u) ->
+        {
+            RepositoryResolver resolver = new IBiblioResolver();
+            resolver.setName(n);
+            resolver.setM2compatible(true);
+            ((IBiblioResolver)resolver).setRoot(u);
+            chain.add(resolver);
+        });
+
+        IvySettings ivySettings = new IvySettings();
+        ivySettings.addConfigured(chain);
+        ivySettings.setDefaultResolver(chain.getName());
+
+        Ivy ivy = Ivy.newInstance(ivySettings);
+
+        Iterator<Map.Entry<String, String>> depIterator = dependencies.entrySet().iterator();
+        boolean hasErrors = false;
+        while (depIterator.hasNext())
+        {
+            Map.Entry<String, String> el = depIterator.next();
+            String[] c = el.getKey().split(":");
+            ModuleRevisionId dependency = ModuleRevisionId.newInstance(c[0], c[1], el.getValue());
+            ResolveReport report = ivy.resolve(dependency, options, false);
+            hasErrors |= report.hasError();
+        }
```

Comment:
Avoid explicit use of iterators unless you need to call `iterator.remove(...)` to remove while iterating without triggering concurrent modification exceptions.
```
for (Map.Entry<String, String> dependency : dependencies.entrySet())
{
    String optionallyQualifiedArtifact = dependency.getKey();
    String version = dependency.getValue();

    String[] parts = optionallyQualifiedArtifact.split(":");
    String groupId = parts.length == 2 ? parts[0] : DEFAULT_GROUP_NAME;
    String artifactId = parts[parts.length - 1];

    ModuleRevisionId reference = ModuleRevisionId.newInstance(groupId, artifactId, version);

    ResolveReport report = ivy.resolve(dependency, options, false);
    hasErrors |= report.hasError();
}
```